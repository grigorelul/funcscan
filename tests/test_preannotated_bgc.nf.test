nextflow_pipeline {

    name "Test pipeline: NFCORE_FUNCSCAN"
    script "main.nf"
    tag "pipeline"
    tag "nfcore_funcscan"
    tag "test_preannotated_bgc"

    test("test_preannotated_bgc_profile") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert new File("$outputDir/pipeline_info/nf_core_funcscan_software_mqc_versions.yml").exists() },
                { assert new File("$outputDir/multiqc/multiqc_report.html").exists() },

                // antiSMASH
                { assert snapshot(
                    path("$outputDir/bgc/antismash/sample_1/css"),
                    path("$outputDir/bgc/antismash/sample_2/css"),
                    path("$outputDir/bgc/antismash/sample_3/css")
                ).match("antismash_css") },                                                                                               // parts of channel: html_accessory_files
                { assert path("$outputDir/bgc/antismash/sample_1/sample_1.gbk").text.contains("##antiSMASH-Data-START##") },                  // channel: gbk_input
                { assert path("$outputDir/bgc/antismash/sample_1/sample_1.zip").exists() },                                                   // channel: zip
                { assert path("$outputDir/bgc/antismash/sample_1/index.html").text.contains("https://antismash.secondarymetabolites.org/") }, // channel: html
                { assert path("$outputDir/bgc/antismash/sample_1/regions.js").text.contains('PROKKA_859') },                                  // channel: json_sideloading
                { assert path("$outputDir/bgc/antismash/sample_1/sample_1.log").text.contains("antiSMASH status: SUCCESS") },                 // channel: log
                { assert path("$outputDir/bgc/antismash/sample_2/sample_2.gbk").text.contains("##antiSMASH-Data-START##") },                  // channel: gbk_input
                { assert path("$outputDir/bgc/antismash/sample_2/sample_2.zip").exists() },                                                   // channel: zip
                { assert path("$outputDir/bgc/antismash/sample_2/index.html").text.contains("https://antismash.secondarymetabolites.org/") }, // channel: html
                { assert path("$outputDir/bgc/antismash/sample_2/regions.js").text.contains('PROKKA_859') },                                  // channel: json_sideloading
                { assert path("$outputDir/bgc/antismash/sample_2/sample_2.log").text.contains("antiSMASH status: SUCCESS") },                 // channel: log
                { assert path("$outputDir/bgc/antismash/sample_3/sample_3.gbk").text.contains("##antiSMASH-Data-START##") },                  // channel: gbk_input
                { assert path("$outputDir/bgc/antismash/sample_3/sample_3.zip").exists() },                                                   // channel: zip
                { assert path("$outputDir/bgc/antismash/sample_3/index.html").text.contains("https://antismash.secondarymetabolites.org/") }, // channel: html
                { assert path("$outputDir/bgc/antismash/sample_3/regions.js").text.contains('NODE_1328_length_3730_cov_3.647347') },          // channel: json_sideloading
                { assert path("$outputDir/bgc/antismash/sample_3/sample_3.log").text.contains("antiSMASH status: SUCCESS") },                 // channel: log

                // DeepBGC
                { assert snapshot(
                    path("$outputDir/bgc/deepbgc/sample_1/sample_1.bgc.gbk"),
                    path("$outputDir/bgc/deepbgc/sample_2/sample_2.bgc.gbk"),
                    path("$outputDir/bgc/deepbgc/sample_3/sample_3.bgc.gbk")
                ).match("deepbgc_bgc_gbk") },                                                                                                      // channel: bgc_gbk
                { assert path("$outputDir/bgc/deepbgc/sample_1/sample_1.antismash.json").text.contains("Putative BGCs predicted using DeepBGC") }, // channel: json
                { assert path("$outputDir/bgc/deepbgc/sample_1/LOG.txt").text.contains('Saved DeepBGC result to:') },                              // channel: log
                { assert path("$outputDir/bgc/deepbgc/sample_1/sample_1.full.gbk").text.contains('1 ttcgccagga gtggcgaagc gatgcgaggt') },          // channel: full_gbk
                { assert path("$outputDir/bgc/deepbgc/sample_2/sample_2.antismash.json").text.contains("Putative BGCs predicted using DeepBGC") }, // channel: json
                { assert path("$outputDir/bgc/deepbgc/sample_2/LOG.txt").text.contains('Saved DeepBGC result to:') },                              // channel: log
                { assert path("$outputDir/bgc/deepbgc/sample_2/sample_2.full.gbk").text.contains('1 aaggggtatg gagcagcgac gtctacccgt') },          // channel: full_gbk
                { assert path("$outputDir/bgc/deepbgc/sample_3/sample_3.antismash.json").text.contains("NODE_1328_length_3730_cov_3.647347") },    // channel: json
                { assert path("$outputDir/bgc/deepbgc/sample_3/LOG.txt").text.contains('Saved DeepBGC result to:') },                              // channel: log
                { assert path("$outputDir/bgc/deepbgc/sample_3/sample_3.full.gbk").text.contains('1 tgaatctgtt ttaaagcaaa ttgatctcgc') },          // channel: full_gbk

                // GECCO
                { assert snapshot(
                    path("$outputDir/bgc/gecco/sample_1/sample_1.genes.tsv"),    // channel: genes
                    path("$outputDir/bgc/gecco/sample_1/sample_1.features.tsv"), // channel: features
                    path("$outputDir/bgc/gecco/sample_2/sample_2.genes.tsv"),    // channel: genes
                    path("$outputDir/bgc/gecco/sample_2/sample_2.features.tsv"), // channel: features
                    path("$outputDir/bgc/gecco/sample_3/sample_3.genes.tsv"),    // channel: genes
                    path("$outputDir/bgc/gecco/sample_3/sample_3.features.tsv")  // channel: features
                ).match("gecco") }
            )
        }
    }
}
