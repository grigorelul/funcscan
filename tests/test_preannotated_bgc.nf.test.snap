{
    "antismash_css": {
        "content": [
            [
                "bacteria.css:md5,39c0ca9cbc64cb824dc958b26b5b4ab8"
            ],
            [
                "bacteria.css:md5,39c0ca9cbc64cb824dc958b26b5b4ab8"
            ],
            [
                "bacteria.css:md5,39c0ca9cbc64cb824dc958b26b5b4ab8"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-24T10:49:00.343547789"
    },
    "deepbgc_bgc_gbk": {
        "content": [
            "sample_1.bgc.gbk:md5,e50e429959e9c4bf0c4b97d9dcd54a08",
            "sample_2.bgc.gbk:md5,effe3cfc91772eb4e4b50ac46f13a941",
            "sample_3.bgc.gbk:md5,41920a93524a1bb32ae1003d69327642"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-24T11:06:00.388012579"
    },
    "gecco": {
        "content": [
            "sample_1.genes.tsv:md5,804af8236a7148baf8919e3acf30947d",
            "sample_1.features.tsv:md5,a84d59fd63e2593dc5872b4f9bb268b2",
            "sample_2.genes.tsv:md5,5a2b20c5c1cd821a2af405229c4c0f78",
            "sample_2.features.tsv:md5,579a27490188f5bc47a4deb4d1d1b8dc",
            "sample_3.genes.tsv:md5,6874723404b3326f0f73e59f03b96837",
            "sample_3.features.tsv:md5,490f98655089b3c73f88b93347cca465"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-24T10:49:00.44526019"
    }
}